### DISCUSSION TITLE
A collection of problems I am encountering with SC

### DISCUSSION THREAD

**Discussion by avataraustin** [brianpetro/obsidian-smart-connections#563] [General]
I recently checked if the updated version was working by querying my notes with a question that I know should return specific relevant info.  It did not give a RAG based answer.  I also had been getting a message asking if I am sure I want to delete 94% of my block level embeddings every time I reopened Obsidian. I ended up finally saying ok, it seemed to still work if I remember correctly.  Fast forward to the latest version I updated to and I decided I wanted to try a local llm configuration with Ollama.  I got it to use local llm for SC query but it was giving me LLM answer only, no RAG.  I tried switching back to Openai for SC queries, I discovered that it too was not using my Notes for the answers it retrieved.  I restarted Obsidian a few times hoping but I ended up picking a bigger embedding model in the hopes that this would solve something. I ended up wasting a day waiting for my computer to finish using Nomic embeddings.  I just tried a SC query and it also gave query results that were not based on my embeddings in any way.  I am getting very frustrated because I have been experiencing a lot of problems before this where my embeddings would get lost and need to be reembedded frequently and at this point it is taking up way too much troubleshooting. I have some collection of screenshots from the dev console but the issues that keep coming up are diverse enough that they might be misleading. as an example, after completing the nomic embeddings it immediately had a pop up message asking if I wanted to start the embedding process even though I had just completed it.  I decided to just quit Obsidian and reopen and when I reopened the message never reappeared, but as stated earlier the SC RAG queries did not work either. As another side note I also keep having SC say there is no Dataview api popup.... not sure what that means. I guess I will at least switch back to a faster embed model so I don't waste another entire day embedding but I hope maybe a lasting solution might be recommended if any of this information is helpful.

this was the pop up I had previously kept getting every time I opened Obsidian:
![Screenshot 2024-04-22 at 9 36 47 AM](https://github.com/brianpetro/obsidian-smart-connections/assets/39737076/bea3788b-9909-497f-8256-125c49d5e4d2)

This was what appeared right after I had actually completed the nomic embeddings:
![Right After completing Nomic embeddings](https://github.com/brianpetro/obsidian-smart-connections/assets/39737076/6d5d14d5-02d4-4d4a-94f2-3aa8468833d3)

This is the Dev console after I did a query(after updating embeddings to Nomic) where I asked a question that should have triggered a specific RAG query, but the results did not give a RAG result:
![Screenshot 2024-04-23 at 9 13 01 AM](https://github.com/brianpetro/obsidian-smart-connections/assets/39737076/2819d30d-c3ee-4db8-b43d-ee811697a92c)

**Comment by brianpetro** [comment]
Hi @avataraustin 

I know it's frustrating when software doesn't work as expected. Thanks for maintaining the patience to share the steps you took troubleshooting and your help in figuring this out 😊

In an effort to maximize the speed of development, I often end up leaving unnecessary logs in the console. That's what's happening with the dataview API log, so this is a log you can ignore unless you utilize dataview and Smart Chat is having trouble rendering your dataview queries. 

Based on your screenshots, it looks like the chat _is_ attempting to utilize your notes as context (RAG), however, it appears that no results are being returned.

Since it seems like the screenshots may not all be from the latest version, I think it's important to note that in the most recent version `v2.1.50` a bug was fixed that may have caused the erroneous "Delete 94% of Block embeddings" notification.

Regarding embedding models, if you are having any issues at all, I highly recommend sticking with the `BGE-micro-v2` model until everything is sorted out. While it may seem "less powerful" than many of the other models, it makes up for this in speed and reliability. Furthermore, the "better" models will often perform the same except in highly sensitive situations, for example trying to differentiate between _many very similar_ documents. Furthermore, with regard to the quality of the Smart Chat retrieval and result in the Smart VIew, there are other ways to improve the results, for example, re-ranking. Adding a re-ranking layer should provide a significant improvement, while changing embedding models will likely result in a marginal improvement at best. And adding re-rankers is one of the features that will be released soon in `v2.2`.

In the meantime, it's important that we get your embedding system working. Besides switching to `BGE-micro-v2`, I suggest turning on the "Embedding File per Note (EXPERIMENTAL)" feature. Managing embeddings is a data-intensive process, and there are many nuances between different computers and how they might impact the embedding storage. So far, some people that have also had the re-embedding issue have found that turning on this new setting fixes things for them. 

![Screenshot 2024-04-23 at 2 00 52 PM](https://github.com/brianpetro/obsidian-smart-connections/assets/1886014/10a09d5f-a32f-494c-8ab7-1b804505385c)

I do want to note that this feature is new and currently has some downsides which I will be addressing. For example, initial start-up times will increase since many more files have to be read to load the system. But, in the long run, this is the format that I believe will become the default as it allows for many other beneficial features and opportunities. So these drawbacks will be addressed in due time.

Thanks again for your patience and understanding as we work together to make Smart Connections a valuable, and preferably also frustration-reducing, software for individuals seeking to improve their lives with AI 🤖
🌴
